src/Compilers/CSharp/Portable/Symbols/Source/SourceLocalSymbol.cs

```diff
@@ -129,7 +140,12 @@ internal Binder Binder
             SyntaxToken identifierToken,
             CSharpSyntaxNode context)
         {
-            return new OutLocalSymbol(containingSymbol, scopeBinder, enclosingBinderOpt, typeSyntax, identifierToken, context);
+            if (typeSyntax.IsVar)
+            {
```

Comment:
The design is consistent in the sense that the factory method decides what specific class should be used to represent particular symbol. The exact criteria doesn't have to match. I think it is useful to avoid using more expensive (memory-wise) objects in obvious scenarios like the one targeted by the condition.